### PR TITLE
Update FindInt in DG

### PR DESCRIPTION
**Pull request by ElijahS67**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -154,6 +154,37 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Find Interest Feature
+
+#### Implementation
+
+The `FindInterestCommandParser` reads the specified interests entered by the student in `FindInterestCommandParser#parse` and passes them as a `PersonContainsInterestPredicate` predicate for execution in `FindInterestCommand`.
+
+Upon execution, the interests of every batchmates will be tested against said predicate in `FindInterestCommand#execute`.
```
Comment (sltsheryl) [inline docs/DeveloperGuide.md:163]: tested against the said predicate

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -154,6 +154,37 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+#### Design considerations:
+
+**Aspect: How `findInt` executes:**
+
+* **Alternative 1 (current choice):** Finds all batchmates whose interests match all interests specified (i.e. a superset of those specified).
+    * Pros: A specific search to find batchmates who have the same interests as the student. (Specifying `tennis` and `baking` will result in a batchmate whose interests are `tennis`, `baking` and `music` to be displayed)
+    * Cons: Search might be too narrow since it excludes batchmates that have some but not all the interests specified. (Specifying `tennis` and `baking` will not result in a batchmate whose only interest is `tennis` to be displayed)
+
+* **Alternative 2:** Finds all batchmates whose interests match at least one of the interests specified.
+    * Pros: A more general search might be useful for finding much more batchmates who share some of the interests as the student.
```
Comment (sltsheryl) [inline docs/DeveloperGuide.md:184]: for finding a greater number of batchmates

**Review by sltsheryl — COMMENTED** [review]
The diagrams look okay

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (ElijahS67) [reply docs/DeveloperGuide.md]: Yup, thanks for the catch!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (ElijahS67) [reply docs/DeveloperGuide.md]: Rewrote it to minimise ambiguity

**Review by sltsheryl — APPROVED** [review]
LGTM

**Pull request merged**